app/supabase-provider.tsx

```diff
@@ -1,8 +1,8 @@
 'use client';
```

Comment:
Since this wraps the rest of the React tree, it will opt the entire app into client rendering. This pattern was a thing to ensure the Supabase client was a singleton, but see my comment below about this no longer being the case with `@supabase/ssr`.

So, if we don't need to use a provider to ensure this is a single instance of Supabase, really, this component is just attaching a listener for auth changes in `useEffect`. Therefore, it doesn't need to be a provider that wraps the entire app, it could just be its own component:

```
export function SupabaseAuthListener() {
  const supabase = createClient()
  
  useEffect(() => {
    // auth listener subscription stuff
  }, [dependency stuff])
}

return null
```

Then this can just be rendered from the root layout, rather than wrapping everything else and opting the entire tree into client rendering.

Reply:
@dijonmusters I know that supabase-provider is being removed, but I was under the impression that server components passed as children of client components are still rendered on the server https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#supported-pattern-passing-server-components-to-client-components-as-props

Reply:
@dalkommatt thank you for all this effort! Did you think about adding this SupabaseAuthListener in? I don't see it in the latest changes for RootLayout. But I think it would be useful for changing themes, or what not once a user is logged in. Thoughts?